pulpcore/cli/file/repository.py

```diff
@@ -42,88 +55,42 @@ def repository(ctx: click.Context, pulp_ctx: PulpContext, repo_type: str) -> Non
 
 
 lookup_options = [href_option, name_option]
+create_options = [
+    click.option("--name", required=True),
+    click.option("--description"),
+    click.option("--remote", callback=_remote_callback),
+]
+update_options = [
+    click.option("--description"),
+    click.option("--remote", callback=_remote_callback),
+]
 
 repository.add_command(list_command())
 repository.add_command(show_command(decorators=lookup_options))
+repository.add_command(create_command(decorators=create_options))
+repository.add_command(update_command(decorators=lookup_options + update_options))
 repository.add_command(destroy_command(decorators=lookup_options))
 repository.add_command(version_command())
 
 
-@repository.command()
-@click.option("--name", required=True)
-@click.option("--description")
-@click.option("--remote")
-@pass_repository_context
-@pass_pulp_context
-def create(
-    pulp_ctx: PulpContext,
-    repository_ctx: PulpRepositoryContext,
-    name: str,
-    description: Optional[str],
-    remote: Optional[str],
-) -> None:
-    repository = {"name": name, "description": description}
-    if remote:
-        remote_href: str = PulpFileRemoteContext(pulp_ctx, entity={"name": remote}).pulp_href
-        repository["remote"] = remote_href
-
-    result = repository_ctx.create(body=repository)
-    pulp_ctx.output_result(result)
-
-
-@repository.command()
-@name_option
-@href_option
-@click.option("--description")
-@click.option("--remote")
-@pass_repository_context
-@pass_pulp_context
-def update(
-    pulp_ctx: PulpContext,
-    repository_ctx: PulpRepositoryContext,
-    description: Optional[str],
-    remote: Optional[str],
-) -> None:
-    repository = repository_ctx.entity
-    repository_href = repository_ctx.pulp_href
-
-    if description is not None:
-        if description == "":
-            # unset the description
-            description = None
-        if description != repository["description"]:
-            repository["description"] = description
-
-    if remote is not None:
-        if remote == "":
-            # unset the remote
-            repository["remote"] = ""
-        elif remote:
-            remote_href: str = PulpFileRemoteContext(pulp_ctx, entity={"name": remote}).pulp_href
-            repository["remote"] = remote_href
-
-    repository_ctx.update(repository_href, body=repository)
-
-
 @repository.command()
 @name_option
 @href_option
-@click.option("--remote")
+@click.option("--remote", callback=_remote_callback)
 @pass_repository_context
 @pass_pulp_context
 def sync(
     pulp_ctx: PulpContext,
```

Comment:
The new generics remove the need for pulp_ctx. This variable isn't even used in the function, so is the pass_pulp_context decorator still necessary?

Reply:
good observation!